sys/net/gnrc/application_layer/dhcpv6/client.c

```diff
@@ -174,6 +174,17 @@ void dhcpv6_client_conf_prefix(unsigned iface, const ipv6_addr_t *pfx,
             gnrc_rpl_root_init(CONFIG_GNRC_RPL_DEFAULT_INSTANCE, &addr, false, false);
         }
     }
+
+    /* start advertising subnet obtained via DHCPv6 */
+    gnrc_ipv6_nib_change_rtr_adv_iface(netif, true);
+}
+
+void dhcpv6_client_conf_done(unsigned iface)
+{
+    gnrc_netif_t *netif = gnrc_netif_get_by_pid(iface);
+
+    /* inform upstream network about the subnet(s) now managed by this router */
+    gnrc_ipv6_nib_change_rtr_adv_rio_iface(netif, true);
 }
```

Comment:
Shouldn't this rather be part of a configuration module like `simple_pd`? Does not seem right, that DHCPv6 _always_ uses this option.

Reply:
With this we can drop e.g.

    ip route add ${PREFIX} via fe80::2 dev ${TAP}

in `start_network.sh`. IMHO it makes sense to always include this so that upstream hosts don't have to manually configure routes.

Reply:
It might make sense for this particular use case, but I have some worries that we might shoot ourselves in the foot for future use cases when we _enforce_ such non-standard behavior. The sending and configuration of RAs has nothing to do with DHCPv6 in general, so the DHCPv6 client _in general_ should not decide its sending behavior. Configurations may do this via hooks, that's why e.g. the deactivation of RAs is in `client_simple_pd`, not in `client` in current master.

Reply:
Do you suggest I should instead register a callback here and move this to `client_simple_pd.c `?

Reply:
I would just move the implementation of `dhcpv6_client_conf_done()` to `client_simple_pd.c`. The name already suggests as much. Maybe add a weak implementation here so `client.c` works without.

Reply:
Ok, made it so.